src/Native/Unix/Common/pal_utilities.h

```diff
@@ -16,6 +16,28 @@
 #include <type_traits>
 #include <unistd.h>
 
+#ifdef DEBUG
+#define assert_err(cond, msg, err) do \
+{ \
+  if(!(cond)) \
+  { \
+    fprintf(stderr, "%s (%d): error %d: %s. %s\n", __FILE__, __LINE__, err, msg, strerror(err)); \
+    assert(cond && "assert_err failed"); \
```

Comment:
This will evaluate cond a second time, which is problematic in this case if it may change its result on a subsequent evaluation (or if it has side effects). Doesn't look like it will be a problem based on your current usage, but as a general facility, it's be best to avoid this. You might make this just be 'false' and then add to the printf a stringified version of cond.